server/src/main/java/org/candlepin/resource/OwnerResource.java

```diff
@@ -1826,67 +1947,7 @@ public void updatePool(@PathParam("owner_key") @Verify(Owner.class) String owner
-
-        // Ignore what the client has specified as the owner, since we already know the owner here
-        newPool.setOwner(currentPool.getOwner());
-
-        /*
-         * These are @JsonIgnored. If a client creates a pool and subsequently
-         * wants to update it , we need to ensure Products are set
-         * appropriately.
-         */
-        newPool.setProduct(currentPool.getProduct());
-        newPool.setDerivedProduct(currentPool.getDerivedProduct());
-
-        // Fill out the new pool with data from the existing pool for values which aren't set
-        // TODO: Abstract this out to something more dedicated if we end up needing this logic elsewhere
-
-        // Start by forcefully setting things we don't allow the client to change. Move these down as
-        // necessary.
-        newPool.setActiveSubscription(currentPool.getActiveSubscription());
-        newPool.setCreatedByShare(currentPool.isCreatedByShare());
-        newPool.setHasSharedAncestor(currentPool.hasSharedAncestor());
-        newPool.setSourceEntitlement(currentPool.getSourceEntitlement());
-        newPool.setSourceStack(currentPool.getSourceStack());
-        newPool.setSourceSubscription(currentPool.getSourceSubscription());
-        newPool.setEntitlements(currentPool.getEntitlements());
-        newPool.setRestrictedToUsername(currentPool.getRestrictedToUsername());
-        newPool.setContractNumber(currentPool.getContractNumber());
-        newPool.setAccountNumber(currentPool.getAccountNumber());
-        newPool.setOrderNumber(currentPool.getOrderNumber());
-        newPool.setConsumed(currentPool.getConsumed());
-        newPool.setExported(currentPool.getExported());
-        newPool.setShared(currentPool.getShared());
-        newPool.setUpstreamPoolId(currentPool.getUpstreamPoolId());
-        newPool.setUpstreamEntitlementId(currentPool.getUpstreamEntitlementId());
-        newPool.setUpstreamConsumerId(currentPool.getUpstreamConsumerId());
-        newPool.setCertificate(currentPool.getCertificate());
-        newPool.setCdn(currentPool.getCdn());
-
-        // Set the fields for those that are null (indicating the user didn't change it)
-        if (newPool.getQuantity() == null) {
-            newPool.setQuantity(currentPool.getQuantity());
-        }
-
-        if (newPool.getStartDate() == null) {
-            newPool.setStartDate(currentPool.getStartDate());
-        }
-
-        if (newPool.getEndDate() == null) {
-            newPool.setEndDate(currentPool.getEndDate());
-        }
-
-        if (newPool.getAttributes() == null) {
-            newPool.setAttributes(currentPool.getAttributes());
-        }
-
-        if (newPool.getBranding() == null) {
-            newPool.setBranding(currentPool.getBranding());
-        }
-
-        // Resolve the bits we may not have set
-        newPool = resolverUtil.resolvePool(newPool);
+        this.populateEntityOnUpdate(newPool, newPoolDTO, currentPool);
```

Comment:
Same here; also not a fan of the three-way merge looking thing here. We're at a point where we don't need to juggle multiple instances of the entities to accomplish this task.